src/components/DayPickerRangeController.jsx

```diff
@@ -174,6 +183,22 @@ export default class DayPickerRangeController extends React.Component {
     this.setDayPickerRef = this.setDayPickerRef.bind(this);
   }
 
+  componentDidMount() {
```

Comment:
Since the determination of which phrase to show is based off of the props and doesn't rely on the actual state of the component (waiting for the input to have focus), you can do this logic in the constructor and set `phrases` to the new object in `this.state`. 

If you put that logic in `componentDidMount`, which gets called after `render`,  setting `state` will trigger another render.

Reply:
(it'd then also need the same logic in `componentWillReceiveProps`)

Reply:
Yep that totally makes sense, I've moved that logic to the constructor and removed `componentDidMount`.  As for `componentWillReceiveProps` that logic was already there.  The bit I added was to make sure that phrase was being initialized at component load time instead of having to wait until props changed.